### PR TITLE
Migrate the HTTP server from Jetty to Armeria

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number.  related https://github.com/apache/skywalking/discussions/8666.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -105,6 +105,10 @@ Release Notes.
 * Activate `receiver-otel#enabledOcRules` receiver with `k8s-cluster,k8s-node,k8s-service,oap,vm` rules on default.
 * Activate `satellite,spring-sleuth` for `agent-analyzer#meterAnalyzerActiveFiles`  on default.
 * Activate `receiver-zabbix` receiver with `agent` rule on default.
+* Replace Http server (GraphQL, agent http protocol) from Jetty with Armeria.
+* [Breaking Change] Remove configuration `restAcceptorPriorityDelta` (env var: `SW_RECEIVER_SHARING_JETTY_DELTA`, `SW_CORE_REST_JETTY_DELTA`).
+* [Breaking Change] Remove configuration `graphql/path` (env var: `SW_QUERY_GRAPHQL_PATH`).
+* [Breaking Change] Remove configuration `restMinThreads` (env var: `SW_CORE_REST_JETTY_MIN_THREADS`, `SW_RECEIVER_SHARING_JETTY_MIN_THREADS`).
```
Comment (wu-sheng) [inline CHANGES.md:111]: With removing all these, is there any thread number or other key configurations of Armeria server should be exposed?

Comment (wu-sheng) [reply CHANGES.md:111]: I noticed MAX_THREAD has been kept.

File: CHANGES.md [hunk 2]
```diff
@@ -105,6 +105,10 @@ Release Notes.
 * Activate `receiver-otel#enabledOcRules` receiver with `k8s-cluster,k8s-node,k8s-service,oap,vm` rules on default.
 * Activate `satellite,spring-sleuth` for `agent-analyzer#meterAnalyzerActiveFiles`  on default.
 * Activate `receiver-zabbix` receiver with `agent` rule on default.
+* Replace Http server (GraphQL, agent http protocol) from Jetty with Armeria.
```
Comment (wu-sheng) [inline CHANGES.md:108]:
```suggestion
* Replace HTTP server (GraphQL, agent HTTP protocol) from Jetty with Armeria.
```

File: CHANGES.md (hunk 1 above)
Comment (wu-sheng) [reply CHANGES.md:111]: Could you make a comparison about size reducing between two binary tars?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModule.java [hunk 3]
```diff
@@ -122,7 +122,7 @@ private void addQueryService(List<Class> classes) {
 
     private void addServerInterface(List<Class> classes) {
         classes.add(GRPCHandlerRegister.class);
-        classes.add(JettyHandlerRegister.class);
+        classes.add(HttpHandlerRegister.class);
```
[the remark this answers is not in the document]
Comment (kezhenxu94) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModule.java]: Done, there are still some `Http***` from third-party classes.

File: CHANGES.md (hunk 1 above)
Comment (kezhenxu94) [reply CHANGES.md]: There is no much reduction in package size, they are about ~132M the same

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**